youtube_dl/extractor/mixcloud.py

```diff
@@ -251,8 +251,12 @@ def _real_extract(self, url):
                 cloudcast_url = cloudcast.get('url')
                 if not cloudcast_url:
                     continue
+                video_id = cloudcast['slug']
```

Comment:
this will break the extraction if `slug` is null.

Reply:
It's on purpose. If the slug is null, a proper video ID can't be generated anyway.

Reply:
the extraction can work just by having the track URL, so the `video_id` is an optional field, and the extraction should proceed even if an entry id has not been found.

Reply:
Okiedoke, I wasn't aware of that. Thanks :)